### PR TITLE
Added cardinality to SQL on xmltable to avoid excessive estimates.

### LINKED ISSUE #752: ORA-07455 when comparing sysrefcursor using exclude.
Hi,

I'm trying to introduce utPLSQL as Unit Testing framework in our PL/SQL components, since we currently don't have real unit testing.

Unfortunately I have a pretty big issue that's blocking me currently:

I'm trying to compare two cursors that only contain 2 rows (or even only one) and exclude one or two fields in the comparison.

Unfortunately, most of the time my database doesn't execute the comparison query in the package ut_compound_data_helper with the error ORA-07455: estimated execution time (2472326 secs), exceeds limit (115200 secs).

Of course it actually only takes miliseconds when it, out of whatever reason, isn't blocked by this.

I can't find out what's the reason for this, I suppose it's the row_number() function in combination with the XMLTYPE? Did you already experience this and have some hints on how to resolve it?

I'm not sure if an Issue created here is the right place for this, but I didn't know where else to ask this. If you could point me to a better place that would also be appreciated!

Thanks a lot,
Johannes

### PR DESCRIPTION
**Pull request by jgebal**
Resolves #752

### CODE REVIEW

File: source/expectations/data_values/ut_compound_data_helper.pkb [hunk 1]
```diff
@@ -95,37 +95,39 @@ create or replace package body ut_compound_data_helper is
...
-             PASSING x.item_data
-             COLUMNS 
-               name     VARCHAR2(4000)  PATH '@xml_valid_name',
-               type      VARCHAR2(4000) PATH '/' 
-             ) xt
-            ) a
+            from (select rownum actual_pos,
+                         xt.name actual_name,
+                         xt.type actual_type
+                    from (select ]'||l_column_filter||q'[ from actual_cols ucd) x,
```
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCl&open=AWYOC5W_vOWp7Yd6ROCl)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCl

File: source/expectations/data_values/ut_compound_data_helper.pkb [hunk 2]
```diff
@@ -95,37 +95,39 @@ create or replace package body ut_compound_data_helper is
...
-             PASSING x.item_data
-             COLUMNS 
-               name     VARCHAR2(4000)  PATH '@xml_valid_name',
-               type      VARCHAR2(4000) PATH '/' 
-             ) xt
-            ) e
+                  select rownum expected_pos,
+                         xt.name expected_name,
+                         xt.type expected_type
+                    from (select ]'||l_column_filter||q'[ from expected_cols ucd) x,
```
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCk&open=AWYOC5W_vOWp7Yd6ROCk)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCk

File: source/expectations/data_values/ut_compound_data_helper.pkb (hunk 1 above)
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCl&open=AWYOC5W_vOWp7Yd6ROCl)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCl

File: source/expectations/data_values/ut_compound_data_helper.pkb (hunk 2 above)
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCk&open=AWYOC5W_vOWp7Yd6ROCk)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCk

File: source/expectations/data_values/ut_compound_data_helper.pkb (hunk 1 above)
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCl&open=AWYOC5W_vOWp7Yd6ROCl)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCl

File: source/expectations/data_values/ut_compound_data_helper.pkb (hunk 2 above)
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCk&open=AWYOC5W_vOWp7Yd6ROCk)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCk

File: source/expectations/data_values/ut_compound_data_helper.pkb (hunk 1 above)
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCl&open=AWYOC5W_vOWp7Yd6ROCl)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCl

File: source/expectations/data_values/ut_compound_data_helper.pkb (hunk 2 above)
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCk&open=AWYOC5W_vOWp7Yd6ROCk)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCk

File: source/expectations/data_values/ut_compound_data_helper.pkb [hunk 3]
```diff
@@ -91,41 +91,46 @@ create or replace package body ut_compound_data_helper is
...
-             COLUMNS 
-               name     VARCHAR2(4000)  PATH '@xml_valid_name',
-               type      VARCHAR2(4000) PATH '/' 
-             ) xt
-            ) e
+                  select /*+ cardinality(xt 100) */
+                         rownum expected_pos,
+                         xt.name expected_name,
+                         xt.type expected_type
+                    from (select ]'||l_column_filter||q'[ from expected_cols ucd) x,
```
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCk&open=AWYOC5W_vOWp7Yd6ROCk)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCk

File: source/expectations/data_values/ut_compound_data_helper.pkb [hunk 4]
```diff
@@ -91,41 +91,46 @@ create or replace package body ut_compound_data_helper is
...
-             COLUMNS 
-               name     VARCHAR2(4000)  PATH '@xml_valid_name',
-               type      VARCHAR2(4000) PATH '/' 
-             ) xt
-            ) a
+            from (select /*+ cardinality(xt 100) */
+                         rownum actual_pos,
+                         xt.name actual_name,
+                         xt.type actual_type
+                    from (select ]'||l_column_filter||q'[ from actual_cols ucd) x,
```
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCl&open=AWYOC5W_vOWp7Yd6ROCl)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCl

Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCl&open=AWYOC5W_vOWp7Yd6ROCl)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCl

File: source/expectations/data_values/ut_compound_data_helper.pkb (hunk 3 above)
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCk&open=AWYOC5W_vOWp7Yd6ROCk)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCk

Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCk&open=AWYOC5W_vOWp7Yd6ROCk)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCk

File: source/expectations/data_values/ut_compound_data_helper.pkb (hunk 4 above)
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCl&open=AWYOC5W_vOWp7Yd6ROCl)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCl

Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCl&open=AWYOC5W_vOWp7Yd6ROCl)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCl

File: source/expectations/data_values/ut_compound_data_helper.pkb (hunk 3 above)
Comment (utplsql-travis) [inline source/expectations/data_values/ut_compound_data_helper.pkb]: Code Smell Code Smell: An illegal character with code point 10 was found in this literal. ([plsql:LiteralsNonPrintableCharactersCheck](https://sonarcloud.io/organizations/utplsql/rules#rule_key=plsql%3ALiteralsNonPrintableCharactersCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=utPLSQL&pullRequest=756&issues=AWYOC5W_vOWp7Yd6ROCk&open=AWYOC5W_vOWp7Yd6ROCk)

[ISSUE_KEY]: AWYOC5W_vOWp7Yd6ROCk

**Comment by jgebal** [comment]
Before adding `cardinality` hint (on 12.2) we see estimate of `683K rows` and `7863M bytes`.
After adding `cardinality` hint estimate is `300 rows` and `3534K bytes`.

Old SQL plan (without hints)
```
SQL_ID  by9zrp3ss2vy8, child number 0
-------------------------------------
with expected_cols as ( select xmltype(:a) as item_data from dual ),    
     actual_cols as ( select xmltype(:b) as item_data from dual ),      
   expected_cols_info as (           select e.*,                  
replace(expected_type,'VARCHAR2','CHAR') expected_type_compare          
   from (                   select rownum expected_pos,                 
         xt.name expected_name,                          xt.type 
expected_type                     from (select ucd.item_data as 
item_data from expected_cols ucd) x,                          xmltable( 
                           '/ROW/*'                            passing 
x.item_data                            columns                          
    name     varchar2(4000)  PATH '@xml_valid_name',                    
          type     varchar2(4000) PATH '/'                          ) 
xt                  ) e         ),         actual_cols_info as (        
   select a.*,                  replace(actual_type,'VARCHAR2','CHAR') 
actual_type_compa
 
Plan hash value: 3270454536
 
-------------------------------------------------------------------------------------------------------------------------------------------------------------
| Id  | Operation                     | Name     | Starts | E-Rows |E-Bytes|E-Temp | Cost (%CPU)| E-Time   | A-Rows |   A-Time   |  OMem |  1Mem | Used-Mem |
-------------------------------------------------------------------------------------------------------------------------------------------------------------
|   0 | SELECT STATEMENT              |          |      1 |        |       |       |  5139K(100)|          |      6 |00:00:00.01 |       |       |          |
|   1 |  SORT ORDER BY                |          |      1 |    683K|  7863M|  5339M|  5139K  (1)| 00:03:21 |      6 |00:00:00.01 |  2048 |  2048 | 2048  (0)|
|*  2 |   VIEW                        |          |      1 |    683K|  7863M|       |  3425K  (1)| 00:02:14 |      6 |00:00:00.01 |       |       |          |
|   3 |    WINDOW SORT                |          |      1 |    683K|  7846M|  5339M|  3425K  (1)| 00:02:14 |     80 |00:00:00.01 | 13312 | 13312 |12288  (0)|
|   4 |     WINDOW SORT               |          |      1 |    683K|  7846M|  5339M|  3425K  (1)| 00:02:14 |     80 |00:00:00.01 | 90112 | 90112 |79872  (0)|
|   5 |      VIEW                     | VW_FOJ_0 |      1 |    683K|  7846M|       |  4718   (1)| 00:00:01 |     80 |00:00:00.01 |       |       |          |
|*  6 |       HASH JOIN FULL OUTER    |          |      1 |    683K|  7846M|    46M|  4718   (1)| 00:00:01 |     80 |00:00:00.01 |  1298K|  1298K| 5815K (0)|
|   7 |        VIEW                   |          |      1 |   8168 |    46M|       |    29   (0)| 00:00:01 |     74 |00:00:00.01 |       |       |          |
|   8 |         VIEW                  |          |      1 |   8168 |    31M|       |    29   (0)| 00:00:01 |     74 |00:00:00.01 |       |       |          |
|   9 |          COUNT                |          |      1 |        |       |       |            |          |     74 |00:00:00.01 |       |       |          |
|  10 |           XMLTABLE EVALUATION |          |      1 |        |       |       |            |          |     74 |00:00:00.01 |       |       |          |
|  11 |        VIEW                   |          |      1 |   8168 |    46M|       |    29   (0)| 00:00:01 |     80 |00:00:00.01 |       |       |          |
|  12 |         VIEW                  |          |      1 |   8168 |    31M|       |    29   (0)| 00:00:01 |     80 |00:00:00.01 |       |       |          |
|  13 |          COUNT                |          |      1 |        |       |       |            |          |     80 |00:00:00.01 |       |       |          |
|  14 |           XMLTABLE EVALUATION |          |      1 |        |       |       |            |          |     80 |00:00:00.01 |       |       |          |
-------------------------------------------------------------------------------------------------------------------------------------------------------------
 
Query Block Name / Object Alias (identified by operation id):
-------------------------------------------------------------
 
   1 - SEL$13      
   2 - SEL$12       / JOINED_COLS@SEL$13
   3 - SEL$12      
   5 - SEL$1        / from$_subquery$_013@SEL$12
   6 - SEL$1       
   7 - SEL$4        / E@SEL$1
   8 - SEL$F3C6CAD4 / E@SEL$4
   9 - SEL$F3C6CAD4
  11 - SEL$8        / A@SEL$1
  12 - SEL$FA3E51C0 / A@SEL$8
  13 - SEL$FA3E51C0
 
Predicate Information (identified by operation id):
---------------------------------------------------
 
   2 - filter(("ACTUAL_POS" IS NULL OR "EXPECTED_POS" IS NULL OR "ACTUAL_TYPE_COMPARE"<>"EXPECTED_TYPE_COMPARE" OR ("A_POS_NN"<>"E_POS_NN" AND 
              "EXPECTED_POS"<>"ACTUAL_POS")))
   6 - access("E"."EXPECTED_NAME"="A"."ACTUAL_NAME")
 
```
New plan (with hints):
```
SQL_ID  1r981zttqtzhw, child number 0
-------------------------------------
with expected_cols as ( select xmltype(:a) as item_data from dual ),    
     actual_cols as ( select xmltype(:b) as item_data from dual ),      
   expected_cols_info as (           select /*+ cardinality(e 100) */ 
e.*,                  replace(expected_type,'VARCHAR2','CHAR') 
expected_type_compare             from (                   select 
rownum expected_pos,                          xt.name expected_name,    
                      xt.type expected_type                     from 
(select ucd.item_data as item_data from expected_cols ucd) x,           
               xmltable(                            '/ROW/*'            
                passing x.item_data                            columns  
                            name     varchar2(4000)  PATH 
'@xml_valid_name',                              type     varchar2(4000) 
PATH '/'                          ) xt                  ) e         ),  
       actual_cols_info as (           select /*+ cardinality(a 100) */ 
a.*,                  repl
 
Plan hash value: 3270454536
 
-------------------------------------------------------------------------------------------------------------------------------------------------------------
| Id  | Operation                     | Name     | Starts | E-Rows |E-Bytes|E-Temp | Cost (%CPU)| E-Time   | A-Rows |   A-Time   |  OMem |  1Mem | Used-Mem |
-------------------------------------------------------------------------------------------------------------------------------------------------------------
|   0 | SELECT STATEMENT              |          |      1 |        |       |       |  2322 (100)|          |      6 |00:00:00.01 |       |       |          |
|   1 |  SORT ORDER BY                |          |      1 |    300 |  3534K|  2408K|  2322   (1)| 00:00:01 |      6 |00:00:00.01 |  2048 |  2048 | 2048  (0)|
|*  2 |   VIEW                        |          |      1 |    300 |  3534K|       |  1567   (1)| 00:00:01 |      6 |00:00:00.01 |       |       |          |
|   3 |    WINDOW SORT                |          |      1 |    300 |  3526K|  2408K|  1567   (1)| 00:00:01 |     80 |00:00:00.01 | 13312 | 13312 |12288  (0)|
|   4 |     WINDOW SORT               |          |      1 |    300 |  3526K|  2408K|  1567   (1)| 00:00:01 |     80 |00:00:00.01 | 90112 | 90112 |79872  (0)|
|   5 |      VIEW                     | VW_FOJ_0 |      1 |    300 |  3526K|       |    58   (0)| 00:00:01 |     80 |00:00:00.01 |       |       |          |
|*  6 |       HASH JOIN FULL OUTER    |          |      1 |    300 |  3526K|       |    58   (0)| 00:00:01 |     80 |00:00:00.01 |  1298K|  1298K| 1352K (0)|
|   7 |        VIEW                   |          |      1 |    100 |   587K|       |    29   (0)| 00:00:01 |     74 |00:00:00.01 |       |       |          |
|   8 |         VIEW                  |          |      1 |    100 |   392K|       |    29   (0)| 00:00:01 |     74 |00:00:00.01 |       |       |          |
|   9 |          COUNT                |          |      1 |        |       |       |            |          |     74 |00:00:00.01 |       |       |          |
|  10 |           XMLTABLE EVALUATION |          |      1 |        |       |       |            |          |     74 |00:00:00.01 |       |       |          |
|  11 |        VIEW                   |          |      1 |    100 |   587K|       |    29   (0)| 00:00:01 |     80 |00:00:00.01 |       |       |          |
|  12 |         VIEW                  |          |      1 |    100 |   392K|       |    29   (0)| 00:00:01 |     80 |00:00:00.01 |       |       |          |
|  13 |          COUNT                |          |      1 |        |       |       |            |          |     80 |00:00:00.01 |       |       |          |
|  14 |           XMLTABLE EVALUATION |          |      1 |        |       |       |            |          |     80 |00:00:00.01 |       |       |          |
-------------------------------------------------------------------------------------------------------------------------------------------------------------
 
Query Block Name / Object Alias (identified by operation id):
-------------------------------------------------------------
 
   1 - SEL$13      
   2 - SEL$12       / JOINED_COLS@SEL$13
   3 - SEL$12      
   5 - SEL$1        / from$_subquery$_013@SEL$12
   6 - SEL$1       
   7 - SEL$4        / E@SEL$1
   8 - SEL$F3C6CAD4 / E@SEL$4
   9 - SEL$F3C6CAD4
  11 - SEL$8        / A@SEL$1
  12 - SEL$FA3E51C0 / A@SEL$8
  13 - SEL$FA3E51C0
 

Predicate Information (identified by operation id):
---------------------------------------------------
 
   2 - filter(("ACTUAL_POS" IS NULL OR "EXPECTED_POS" IS NULL OR "ACTUAL_TYPE_COMPARE"<>"EXPECTED_TYPE_COMPARE" OR ("A_POS_NN"<>"E_POS_NN" AND 
              "EXPECTED_POS"<>"ACTUAL_POS")))
   6 - access("E"."EXPECTED_NAME"="A"."ACTUAL_NAME")
 
```

**Pull request merged**